### PR TITLE
refactor: consistent import order

### PR DESCRIPTION
**Pull request by wass3r**
just testing the waters to see what folks think. maybe i'm not the only one bothered by import order being all over the place? i used [gci](https://github.com/daixiang0/gci) via ~`gci write --custom-order -s standard -s default -s blank -s dot -s alias -s "prefix(github.com/go-vela)" .`~ `gci write --custom-order -s standard -s default -s blank -s dot -s "prefix(github.com/go-vela)" .` (UPDATED - see comments) to format all of these. it's a little opinionated. was mostly going for std lib, third party, and our own imports last. we can add this to golangci as a check, but i didn't do it here (yet) to get feedback first.

if we like it, we can apply to all repos.

### CODE REVIEW

**Comment by plyr4** [comment]
I like it

**Comment by wass3rw3rk** [comment]
@plyr4 do you like it enough for it to be a linter rule that will _gently_ remind folks that their import order might be ... out of order? golangci-lint does support `gci`.

**Comment by plyr4** [comment]
@wass3rw3rk yeah it feels like a reasonable linter rule, especially if we can add it to our golangci-lint action

**Comment by plyr4** [comment]
can we supply the configurations you used in 
```
gci write --custom-order -s standard -s default -s blank -s dot -s alias -s "prefix(github.com/go-vela)"
```
as a file so we can commit the parameters to source?

**Comment by wass3r** [comment]
@plyr4 see https://github.com/go-vela/server/pull/1077/commits/83d42075609f15c89cf56b99f577bdb7cd5a1910

with this in place you could run `golangci-lint run --fix` and it will fix any violation if the linter supports it. that is the case for `gci`. we can also add it in CONTRIBUTING.md, if that's what you were thinking?

**Comment by plyr4** [comment]
@wass3rw3rk oh yeah exactly! i was thinking of it from "run gci with config" and not "run golangci-lint with gci config" that is great

**Review by ecrupper — DISMISSED** [review]
Very cool. I am pro linter rule too

File: .golangci.yml [hunk 1]
```diff
@@ -32,6 +32,19 @@ linters-settings:
...
+  # gci write --custom-order -s standard -s default -s blank -s dot -s alias -s "prefix(github.com/go-vela)" .
+  gci:
+    custom-order: true
+    sections:
+      - standard
+      - default
+      - blank
+      - dot
+      - alias
+      - prefix(github.com/go-vela)
```
Comment (cognifloyd) [inline .golangci.yml:46]:
```suggestion
  # gci write --custom-order -s standard -s default -s blank -s dot -s "prefix(github.com/go-vela)" .
  gci:
    custom-order: true
    sections:
      - standard
      - default
      - blank
      - dot
      - prefix(github.com/go-vela)
```
Speaking of opinions: I don't like separating aliased imports from the other groups. What about skipping that section?

Comment (wass3r) [reply .golangci.yml:46]: I'm good with that!

File: compiler/template/native/render_test.go [hunk 2]
```diff
@@ -6,9 +6,10 @@ import (
 	"os"
 	"testing"
```
[the remark this answers is not in the document]
Comment (wass3r) [reply compiler/template/native/render_test.go]: outdated

File: compiler/template/native/render_test.go [hunk 3]
```diff
@@ -6,9 +6,10 @@ import (
...
 
-	goyaml "github.com/buildkite/yaml"
 	"github.com/google/go-cmp/cmp"
 
+	goyaml "github.com/buildkite/yaml"
```
[the remark this answers is not in the document]
Comment (wass3r) [reply compiler/template/native/render_test.go]: outdated

File: compiler/template/native/render.go [hunk 4]
```diff
@@ -7,13 +7,12 @@ import (
 	"fmt"
 	"text/template"
 
-	"github.com/go-vela/types/raw"
+	"github.com/Masterminds/sprig/v3"
+	"github.com/buildkite/yaml"
 
 	types "github.com/go-vela/types/yaml"
```
[the remark this answers is not in the document]
Comment (wass3r) [reply compiler/template/native/render.go]: outdated

File: .golangci.yml (hunk 1 above)
Comment (wass3r) [reply .golangci.yml]: diff here https://github.com/go-vela/server/pull/1077/commits/dd8726309b37301b94c8d2f39c497e9278a1a734 with alias taken out

**Review by cognifloyd — APPROVED** [review]

**Review by timhuynh94 — APPROVED** [review]

**Review by ecrupper — APPROVED** [review]

**Pull request merged**